ddtrace/internal/utils/http.py

```diff
@@ -38,6 +39,19 @@ def strip_query_string(url):
     return h + fs + f
 
 
+def redact_query_string(query_string, query_string_obfuscation_pattern):
+    return query_string_obfuscation_pattern.sub("<redacted>", compat.to_unicode(query_string))
+
+
+def redact_url(url, query_string_obfuscation_pattern):
+    # type: (str, re.Pattern) -> str
+    hqs, fs, f = url.partition("#")
+    h, qss, qs = hqs.partition("?")
+    if qs:
+        return h + qss + redact_query_string(qs, query_string_obfuscation_pattern) + fs + f
+    return url
```

Comment:
What is the expected distribution of the `query_string`\`url` to redact? Do we expect some to occur frequently? If so, perhaps we could memoize these helpers? `set_http_meta` tends to be called on hot paths so the more CPU cycle we can shave off here the better.

Reply:
Probably mostly worth measuring it first, then we can decide if the performance overhead is too much or not.

No need to add any complexity if none is needed.